### DISCUSSION TITLE
I made an export in preparation for a deletion, didn't help when a deletion of the server was done.

### DISCUSSION THREAD

**Discussion by Croworbit** [Tyrrrz/DiscordChatExporter#1280] [Q&A] [answered]
About a year ago to the month, I made a full export of a discord server, in particular one of its channels with a lot of media.
today, the server was silently deleted without warning.
I had prepeared for this very moment, but upon review of the archive, all the media just says 'attachment' and clicking it links to the embed within discord, the one that no longer exists
so all of it is gone. 

the whole point I tried to make a backup is that I could have something locally when the online version inevitably went down

I didn't want to be that person who shows up and complains because they didn't think of making a backup before it was gone, so I made a backup. and now I might as well be in that position for all that was saved.
is there anything I can do at this point?

 
how can I be sure this never happens again?

**Comment by 96-LB** [comment, marked as answer]
> is there anything I can do at this point?

Try copying the media links into a channel on Discord. If the server deletion was recent and media hasn't yet been removed from Discord's database, this will refresh the CDN signature and display the media anew. If the attachments have already been fully deleted, then I'm afraid there's nothing you can do.

> how can I be sure this never happens again?

If you turn on the "Download assets" setting in the export menu on the GUI version, or use the `--media` flag on the CLI version, all attachments and referenced media will be downloaded locally instead of linking to Discord's servers. That way you have a local copy if things disappear.

**Comment by Croworbit** [comment]
> Try copying the media links into a channel on Discord. 

in what way should I do this? log into discord on the browser, then paste any of those  links at the address bar at the top? or do I have to edit the link or something else?

**Reply by 96-LB** [reply]
Simply send the link as a message. The image will then display, unless it's deleted.

*** 

The reason this works:

In the past year, CDN links have become signed, which means they're only visible if there's a valid an unexpired signature. For example, if I copy media from a channel, it looks like this:

https://cdn.discordapp.com/attachments/463454211791912980/788872466739101737/EWzZaCcXQAAhfLV.png?ex=66cf8f3c&is=66ce3dbc&hm=6df02d3f9a238678bf7d84fe3120aa89d0fa20cf291f4727f1bf1cf4d00c0b37&

Everything after the ? is part of the signature. Only Discord can generate signatures, and they expire after 24 hours. This is because Discord doesn't want people using the app as a filehost, so Discord links can only be viewed indefinitely within the actual app. Any links viewed within the app have their signatures automatically refreshed, so by sending the link as a message within Discord you should be able to view and then download the files.

**Reply by Croworbit** [reply]
yeah, that didn't work either. I just get blank black pages and the images don't embed in a channel either. Thanks for telling me how to do all of this, really
even if it didn't work out for me

I'm pretty frustrated and bummed that everything happened this way, I really tried to make an effort.
at least I know for next time, but I had really hoped to avoid this

thanks again
